pyproject.toml

```diff
@@ -75,6 +75,7 @@ spacy = [
 test = [
     "pytest>=5.4.3",
     "pytest-cov>=2.6.1",
+    "ruff~=0.4.7",
```

Comment:
Is there a specific reason you used the ~= operator here? I can imagine using it likely makes dependency management more stable but I like using >= to quickly find out when new major/minor versions are released and things break 😅 That way, I do not have to make changes to support new releases and only change when something breaks (which people are more likely to communicate rather than when things work).

Reply:
I ended up going with this version constraint because if, in a PR, someone were to install a newly released version, it may introduce new rules which reformat/start erroring on code unrelated to the PR.

Arguably the linter version can be kept constant forever, but it can be safely incremented in a separate PR every so often to avoid this.